client/posts/embed.ts

```diff
@@ -60,29 +67,60 @@ function formatNoEmbed(type: provider): (s: string) => string {
 	}
 }
 
+// fetcher for the HookTube provider
+function fetchHookTube(): (el: Element) => Promise<void> {
+	return async (el: Element) => {
```

Comment:
I'm fixing it myself, but just FYI, there really is no point in this function factory, since you don't use any context parameters. Just a simple async function is enough.